client/src/app/shared/components/status-icon/status-icon.tsx

```diff
@@ -80,44 +97,42 @@ export const StatusIcon: React.FC<IStatusIconProps> = ({
     <Icon
       color={isDisabled ? disabledColor.value : iconList[status].color.value}
       className={className}
+      isInline
     />
   );
 
-  let label: string;
-  switch (status) {
-    case "Canceled":
-      label = t("terms.canceled");
-      break;
-    case "Completed":
-      label = t("terms.completed");
-      break;
-    case "Failed":
-      label = t("terms.failed");
-      break;
-    case "InProgress":
-      label = t("terms.inProgress");
-      break;
-    case "NotStarted":
-      label = t("terms.notStarted");
-      break;
-    case "Scheduled":
-      label = t("terms.scheduled");
-      break;
-    default:
-      label = t("terms.unknown");
-      break;
+  if (!label) {
+    switch (status) {
+      case "Canceled":
+        label = t("terms.canceled");
+        break;
```

Comment:
This is some Tackle v1 code. Would be great to implement both of your suggestions here. +1 to getting rid of switch statements & +1 to use the null coalescing operator. I wonder if it would be worth adding some stricter TS/ eslint rules during this downtime between features.